tracer/src/Datadog.Trace/Telemetry/TelemetryController.cs

```diff
@@ -26,7 +26,7 @@ internal class TelemetryController : ITelemetryController
         private readonly TelemetryTransportManager _transportManager;
         private readonly TimeSpan _flushInterval;
         private readonly TimeSpan _heartBeatInterval;
-        private readonly TaskCompletionSource<bool> _tracerInitialized = new();
+        private readonly TaskCompletionSource<bool> _tracerInitialized = new(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
Yikes 😬 

I wonder if we need/should do the same here too:
- https://github.com/DataDog/dd-trace-dotnet/blob/6141da4f19f9146dffa0eb8b3659f38ba974727b/tracer/src/Datadog.Trace/Logging/DirectSubmission/Sink/PeriodicBatching/BatchingSink.cs#L29
- https://github.com/DataDog/dd-trace-dotnet/blob/92cfb890998f14321470d9a51d9f99b0ed439a05/tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs#L443
- https://github.com/DataDog/dd-trace-dotnet/blob/c1d95ca60d1187bcea79905f0168ad49086e5507/tracer/src/Datadog.Trace/Util/AsyncManualResetEvent.cs#L105

Reply:
In which case, there's also these, but I thought they looked less suspect 😉 

- https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Agent/MovingAverageKeepRateCalculator.cs#L31
- https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Agent/StatsAggregator.cs#L50

Reply:
The one in StatsAggregator is OK:

```
        public Task DisposeAsync()
        {
            _discoveryService.RemoveSubscription(HandleConfigUpdate);
            _processExit.TrySetResult(true);
            return _flushTask;
        }
```

Even if the continuation is inlined in `_processExit.TrySetResult`, it's not a problem since we expect the caller to await `_flushTask` (and so wait for the completion either way)